core/src/main/java/uk/gov/justice/services/core/interceptor/InterceptorContext.java

```diff
@@ -0,0 +1,100 @@
+package uk.gov.justice.services.core.interceptor;
+
+import static uk.gov.justice.services.core.interceptor.ContextPayload.contextPayload;
+import static uk.gov.justice.services.core.interceptor.ContextPayload.contextPayloadWith;
+import static uk.gov.justice.services.core.interceptor.ContextPayload.copyWithEnvelope;
+
+import uk.gov.justice.services.messaging.JsonEnvelope;
+
+import java.util.Optional;
+
+import javax.enterprise.inject.spi.InjectionPoint;
+
+public class InterceptorContext {
+
+    private final ContextPayload input;
+    private final ContextPayload output;
+    private final InjectionPoint injectionPoint;
+
+    private InterceptorContext(final ContextPayload input, final ContextPayload output, final InjectionPoint injectionPoint) {
+        this.input = input;
+        this.output = output;
+        this.injectionPoint = injectionPoint;
+    }
+
+    private InterceptorContext(final JsonEnvelope input, final InjectionPoint injectionPoint) {
```

Comment:
I know this is a private method, but do you think it is worth documenting that without output specified it uses the input as an output? It is kind of a fundamental part?